### PR TITLE
Introduce a tooltip to the activity icon providing details regarding the user's most recent interaction. 

### PR DESCRIPTION
**Pull request by MrKalach**
Introduce a tooltip to the activity icon providing details regarding the user's most recent interaction. Time resolution is set to days

Couple screenshots:
image
image

### CODE REVIEW

File: backend/src/managers/UserManager.ts [hunk 1]
```diff
@@ -696,4 +697,14 @@ export default class UserManager {
     async getPublicKey(userId: number) {
         return this.userCache.getPublicKey(userId);
     }
+
+    async getUserVisitedDaysAgo(userId: number): Promise<number> {
+        const lastVisited = await this.userRepository.getUserLastVisited(userId);
+
+        const daysAgo =  lastVisited ?
+            moment().diff(lastVisited, "days") : 0;
```
Comment (Aivean) [inline backend/src/managers/UserManager.ts]: lint

File: frontend/src/Pages/UserPage.module.scss [hunk 2]
```diff
@@ -28,8 +28,9 @@
     }
 }
 .active {
+  position: relative;
   padding-right: 20px;
-  pointer-events: none;
+  //pointer-events: none;
```
Comment (Aivean) [inline frontend/src/Pages/UserPage.module.scss]: Please remove dead code.

File: backend/src/managers/UserManager.ts [hunk 3]
```diff
@@ -18,6 +18,7 @@ import {FeedSorting} from '../api/types/entities/common';
 import TheParser from '../parser/TheParser';
 import {UserCache} from './UserCache';
 import {aesDecryptFromBase64, aesEncryptToBase64} from '../parser/CryptoUtils';
+import moment from 'moment';
```
Comment (Aivean) [inline backend/src/managers/UserManager.ts]: Moment.js is a transitive dependency. Either add it explicitly as a dependency into `package.json`, or, better yet, just implement it as a formula, like `Math.abs(date2 - date1) / (3600 * 24 * 1000)`.

File: frontend/src/Pages/UserPage.module.scss [hunk 4]
```diff
@@ -84,3 +85,42 @@
         margin-right: 4px;
     }
 }
+
+.active .tooltipText {
```
Comment (Aivean) [inline frontend/src/Pages/UserPage.module.scss:88]: Tooltip looks nice, but please consolidate the color and border thickness with other menus, like:
image

https://github.com/spaceshelter/orbitar/blob/3c8d12aae420a65adde04dd0f2c22732439e5ebf/frontend/src/Components/PostComponent.module.scss#L116

File: frontend/src/Pages/UserPage.tsx [hunk 5]
```diff
@@ -88,6 +88,13 @@ export const UserPage = observer(() => {
             refreshProfile();
         };
 
+        const daysAgoToStr = (daysAgo: number) => {
+            const titles = ['день', 'дня', 'дней'];
+            const cases = [2, 0, 1, 1, 1, 2];
+            return daysAgo === 0 ? 'сегодня'
+                : `${daysAgo} ${titles[daysAgo % 100 > 4 && daysAgo % 100 < 20 ? 2 : cases[daysAgo % 10 < 5 ? daysAgo % 10 : 5]]} назад`;
```
Comment (Aivean) [inline frontend/src/Pages/UserPage.tsx]: image

Two notes:
1. Multiline text breaks the layout of the tooltip
2. Precise number of days makes little sense for anything over ≈5. Consider using 
`moment.duration.humanize`:

```
`${moment.duration(daysAgo, 'days').humanize(true)} назад`
```
with different threshold for days, weeks, months, years

File: backend/src/managers/UserManager.ts (hunk 1 above)
Comment (Aivean) [inline backend/src/managers/UserManager.ts]: returning `0` for users who never visited the site could be confusing. Maybe worth the effort to return `undefined`/`null` in this case and handle it in UI.

**Review by Aivean — CHANGES_REQUESTED** [review]
Overall looks good, but there are some minor notes.

File: frontend/src/Pages/UserPage.module.scss (hunk 4 above)
Comment (MrKalach) [reply frontend/src/Pages/UserPage.module.scss:88]: Fixed 
image

File: backend/src/managers/UserManager.ts (hunk 1 above)
Comment (MrKalach) [reply backend/src/managers/UserManager.ts]: done

File: frontend/src/Pages/UserPage.module.scss (hunk 2 above)
Comment (MrKalach) [reply frontend/src/Pages/UserPage.module.scss]: done

File: backend/src/managers/UserManager.ts (hunk 3 above)
Comment (MrKalach) [reply backend/src/managers/UserManager.ts]: Fixed, calculating days directly, without the lib

File: frontend/src/Pages/UserPage.tsx (hunk 5 above)
Comment (MrKalach) [reply frontend/src/Pages/UserPage.tsx]: both 1 & 2 fixed
image

File: backend/src/managers/UserManager.ts (hunk 1 above)
Comment (MrKalach) [reply backend/src/managers/UserManager.ts]: Fixed, now we do not show any tooltip, if there are no user records in activity db.

**Review by Aivean — APPROVED** [review]
LGTM

**Pull request merged**